src/overlays/actors/ovl_Bg_Astr_Bombwall/z_bg_astr_bombwall.c

```diff
@@ -28,59 +36,194 @@ const ActorInit Bg_Astr_Bombwall_InitVars = {
+
+void func_80C09ED0(ColliderTrisInit* arg0, Vec3f* arg1, Vec3s* arg2, ColliderTris* arg3) {
+    s32 i;
+    s32 j;
+    Vec3f sp54[3];
+
+    Matrix_RotateYS(arg2->y, MTXMODE_NEW);
+    Matrix_RotateXS(arg2->x, MTXMODE_APPLY);
+    Matrix_RotateZS(arg2->z, MTXMODE_APPLY);
+
+    for (i = 0; i < arg0->count; i++) {
+        for (j = 0; j < 3; j++) {                                          // https://decomp.me/scratch/JrEnl
+            Matrix_MultVec3f(&(arg0->elements + i)->dim.vtx[j], &sp54[j]); //! FAKE MATCH:
+            Math_Vec3f_Sum(&sp54[j], arg1, &sp54[j]);
+        }
+        Collider_SetTrisVertices(arg3, i, &sp54[0], &sp54[1], &sp54[2]);
+    }
+}
+
+void BgAstrBombwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ColliderTris* temp_a1;
+    BgAstrBombwall* this = THIS;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init((DynaPolyActor*)&this->dyna.actor, 1);
+    DynaPolyActor_LoadMesh(globalCtx, (DynaPolyActor*)&this->dyna.actor, &D_06002498);
+    temp_a1 = &this->unk160;
+    Collider_InitTris(globalCtx, temp_a1);
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x7F) != 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    this->dyna.actor.flags |= 0x10000000;
+    if (Collider_SetTris(globalCtx, temp_a1, &this->dyna.actor, &sTrisInit, &this->unk180) == 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    func_80C09ED0(&sTrisInit, &this->dyna.actor.world.pos, &this->dyna.actor.shape.rot, temp_a1);
+    SubS_FillCutscenesList(&this->dyna.actor, &this->unk238, 1);
+    func_80C0A378(this);
+}
+
+void BgAstrBombwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgAstrBombwall* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80C0A120(BgAstrBombwall* thisx, GlobalContext* globalCtx) {
+    s32 var_s1;
+    Vec3f spC8;
+    Vec3f spBC;
+    s16 var_v0;
+    Vec3f spAC;
+    f32 temp_fv0;
+    Matrix_RotateYS(thisx->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    for (var_s1 = 0; var_s1 < 0x1E; var_s1++) {
+        spC8.x = Rand_Centered() * 140.0f;
+        spC8.y = Rand_ZeroOne() * 200.0f;
+        spC8.z = 0.0f;
+        Matrix_MultVec3f(&spC8, &spBC);
+        Math_Vec3f_Sum(&thisx->dyna.actor.world.pos, &spBC, &spBC);
+        func_800BBFB0(globalCtx, &spBC, 50.0f, 2, Rand_ZeroOne() * 120.0f + 20.0f, Rand_ZeroOne() * 240.0f + 20.0f, 0);
+        spAC.x = Rand_ZeroOne() * 2.5f;
+        spAC.y = (Rand_ZeroOne() * 2.5f) + 1.0f;
+        spAC.z = Rand_ZeroOne() * 2.5f;
+        temp_fv0 = Rand_ZeroOne();
+
+        if (temp_fv0 < 0.2f) {
+            var_v0 = 0x60;
+        } else if (temp_fv0 < 0.6f) {
+            var_v0 = 0x40;
+        } else {
+            var_v0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &spBC, &spAC, &spBC, -0x104, var_v0, 0x14, 0, 0, 0xA, 0, 0, 0x32, -1, 0x267,
+                             &D_06002178);
```

Comment:
```suggestion
        EffectSsKakera_Spawn(globalCtx, &spBC, &spAC, &spBC, -0x104, var_v0, 0x14, 0, 0, 0xA, 0, 0, 0x32, -1, 0x267,
                             object_astr_obj_DL_002178);
```
This is getting the address of the object to spawn particles, you can replace it with this, but you will need to be sure to add `#include objects/object_astr_obj/object_astr_obj.h` at the top of the file for this to work